src/project/manifest/mod.rs

```diff
@@ -483,6 +488,51 @@ impl Manifest {
             .any(|f| f.pypi_dependencies.is_some())
     }
 
+    /// Returns what pypi mapping configuration we should use.
+    /// It can be a custom one  in following format : conda_name: pypi_name
+    /// Or we can use our self-hosted
+    pub fn pypi_name_mapping_source(&self) -> miette::Result<MappingSource> {
+        match self.parsed.project.conda_pypi_map.clone() {
+            Some(url) => {
+                let config = Config::load_global();
+
+                // transform user defined channels into rattler::Channel
+                let channels = url
+                    .keys()
+                    .map(|channel_str| {
+                        Channel::from_str(channel_str, config.channel_config())
+                            .map(|channel| (channel_str, channel.canonical_name()))
+                    })
+                    .collect::<Result<Vec<_>, _>>()
+                    .into_diagnostic()?;
+
+                let mapping = channels
+                        .iter()
+                        .map(|(channel_str, channel)| {
+
+                            let mapping_location = url.get(*channel_str).unwrap();
+
+                            let url_or_path = match Url::parse(mapping_location) {
+                                Ok(url) => MappingLocation::Url(url),
+                                Err(err) => {
+                                    if let ParseError::RelativeUrlWithoutBase = err {
+                                        MappingLocation::Path(PathBuf::from(mapping_location))
+                                    } else {
+                                        miette::bail!("Could not convert {mapping_location} to neither URL or Path")
+                                    }
+                                }
+                            };
+
+                            Ok((channel.clone(), url_or_path))
+                        })
+                        .collect::<miette::Result<HashMap<ChannelName, MappingLocation>>>()?;
+
+                Ok(MappingSource::Custom { mapping })
```

Comment:
I think it's better to also warn the user if the have a mapping to `conda-flibediefloep` that it tells them that that channel doesn't exist.

Reply:
<img width="890" alt="Screenshot 2024-04-08 at 14 49 51" src="https://github.com/prefix-dev/pixi/assets/25200469/98ccacb3-6d73-43bc-8583-af9faa059186">
Changed the logic a little so know it will print a warning if defined channel is not present in project.channels